c/tests/test_trees.c

```diff
@@ -380,76 +380,94 @@ verify_tree_next_prev(tsk_treeseq_t *ts)
 static void
 verify_tree_diffs(tsk_treeseq_t *ts)
 {
-    int ret;
+    int ret, valid_tree;
     tsk_diff_iter_t iter;
     tsk_tree_t tree;
     tsk_edge_list_node_t *record;
     tsk_edge_list_t records_out, records_in;
-    size_t num_nodes = tsk_treeseq_get_num_nodes(ts);
-    size_t j, num_trees;
-    double left, right;
+    tsk_size_t num_nodes = tsk_treeseq_get_num_nodes(ts);
+    tsk_size_t j, num_trees, include_terminal;
+    double lft, rgt;
     tsk_id_t *parent = malloc(num_nodes * sizeof(tsk_id_t));
     tsk_id_t *child = malloc(num_nodes * sizeof(tsk_id_t));
     tsk_id_t *sib = malloc(num_nodes * sizeof(tsk_id_t));
+    tsk_flags_t options;
 
     CU_ASSERT_FATAL(parent != NULL);
     CU_ASSERT_FATAL(child != NULL);
     CU_ASSERT_FATAL(sib != NULL);
-    for (j = 0; j < num_nodes; j++) {
-        parent[j] = TSK_NULL;
-        child[j] = TSK_NULL;
-        sib[j] = TSK_NULL;
-    }
-    ret = tsk_diff_iter_init(&iter, ts);
-    CU_ASSERT_EQUAL_FATAL(ret, 0);
-    ret = tsk_tree_init(&tree, ts, 0);
-    CU_ASSERT_EQUAL_FATAL(ret, 0);
-    ret = tsk_tree_first(&tree);
-    CU_ASSERT_EQUAL_FATAL(ret, 1);
-    tsk_diff_iter_print_state(&iter, _devnull);
-
-    num_trees = 0;
-    while ((ret = tsk_diff_iter_next(&iter, &left, &right, &records_out, &records_in))
-           == 1) {
-        tsk_diff_iter_print_state(&iter, _devnull);
-        num_trees++;
-        /* Update forwards */
-        for (record = records_out.head; record != NULL; record = record->next) {
-            parent[record->edge.child] = TSK_NULL;
-        }
-        for (record = records_in.head; record != NULL; record = record->next) {
-            parent[record->edge.child] = record->edge.parent;
-        }
-        /* Now check against the sparse tree iterator. */
+    for (include_terminal = 0; include_terminal <= 1; include_terminal++) {
```

Comment:
For functions with options I've been using the pattern at https://github.com/tskit-dev/tskit/blob/master/c/tests/test_tables.c#L782 as I think it is clearer what's going on.

Reply:
Sorry - is that the right link @benjeffery - I can't see anything about options there?

Reply:
Yeah, that's the right link - basically call the test function with each option. In the link:
```
    test_edge_table_with_options(0);
    test_edge_table_with_options(TSK_NO_METADATA);
```

Reply:
Oh, I see. Sorry - I was being dense. I agree, much clearer.